### PR TITLE
feat(yaml loader): add loader

### PR DESCRIPTION
**Pull request by AugustinMauroy**
- support `yml` and `yaml` ext
- output a object from the load hooks
- use jsdoc to simplify the DX
- 100 test coverage

#34

### CODE REVIEW

File: packages/yaml/README.md [hunk 1]
```diff
@@ -0,0 +1,36 @@
+# Nodejs Loaders: YAML
+
+<img src="https://raw.githubusercontent.com/JakobJingleheimer/nodejs-loaders/refs/heads/main/logo.svg" height="100" width="100" alt="@node.js loaders logo" />
+
+[![npm version](https://img.shields.io/npm/v/nodejs-loaders/yaml.svg)](https://www.npmjs.com/package/nodejs-loaders/yaml)
+![size](https://img.shields.io/github/languages/code-size/JakobJingleheimer/nodejs-loaders/yaml)
+
+**Environment**: test, development, maybe production (but carefully)
```
Comment (JakobJingleheimer) [inline packages/yaml/README.md:8]:
```suggestion
**Environment**: test, development
```

Loaders should not be used in production

**Review by JakobJingleheimer — APPROVED** [review]

**Comment by JakobJingleheimer** [comment]
The root `package.json` should be updated to mention this new loader :)

File: packages/yaml/package.json [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+		"node.js",
+		"yaml"
+	],
+	"engines": {
+		"node": ">20"
+	},
+	"dependencies": {
+		"@nodejs-loaders/parse-filename": "1.0.0",
+		"yaml": "^2.6.0"
+	}
```
Comment (JakobJingleheimer) [inline packages/yaml/package.json:23]:
```suggestion
	},
	"repository": {
    "url": "https://github.com/JakobJingleheimer/nodejs-loaders"
  }
```

Comment (AugustinMauroy) [reply packages/yaml/package.json:23]: ✨Well done✨

**Review by JakobJingleheimer — COMMENTED** [review]
Still needs to be added to the root README :)

Also, for yaml's readme, see https://github.com/JakobJingleheimer/nodejs-loaders/pull/44 for updated badges :)

**Comment by AugustinMauroy** [comment]
Yup it's on my local changé but not push

**Comment by AugustinMauroy** [comment]
> Still needs to be added to the root README :)

it's done. or I missed something

File: packages/yaml/package.json [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+	"engines": {
+		"node": ">20"
+	},
+	"dependencies": {
+		"@nodejs-loaders/parse-filename": "1.0.0",
+		"yaml": "^2.6.0"
+	},
+    "repository": {
+        "url": "https://github.com/JakobJingleheimer/nodejs-loaders"
+    }
```
Comment (JakobJingleheimer) [inline packages/yaml/package.json:26]: indentation is a little off

```suggestion
  "repository": {
    "url": "https://github.com/JakobJingleheimer/nodejs-loaders"
  }
```

**Review by JakobJingleheimer — APPROVED** [review]
🙌

File: packages/yaml/package.json (hunk 3 above)
Comment (AugustinMauroy) [reply packages/yaml/package.json:26]: may be merge biome before this one to fix that x'D

**Review by JakobJingleheimer — APPROVED** [review]

**Comment by JakobJingleheimer** [comment]
Is this ready to merge?

**Comment by AugustinMauroy** [comment]
> Is this ready to merge?

We said that we wait biome first to format/lint this package then merge then publish

https://github.com/nodejs-loaders/nodejs-loaders/pull/36#discussion_r1874746837

**Pull request merged**